cluster-autoscaler/processors/scaledowncandidates/scale_down_candidates_delay_processor_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaledowncandidates
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	testprovider "k8s.io/autoscaler/cluster-autoscaler/cloudprovider/test"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+)
+
+func TestGetScaleDownCandidates(t *testing.T) {
+	n1 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "n1",
+		},
+	}
+
+	n2 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "n2",
+		},
+	}
+
+	n3 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "n3",
+		},
+	}
+
+	// Case 1
```

Comment:
Can you make this test table based instead of explaining each case in comments?

Reply:
Table tests make the tests a little bit complicated (with the advantage of less duplication) in this case because I want to tweak the processor a bit (specific to a test) before running the test. That being said, here's my attempt: [`cea3044` (#5729)](https://github.com/kubernetes/autoscaler/pull/5729/commits/cea304439820d1e85b9245231839be510fdfd96f)

Reply:
Thanks, that looks better. I'd just consider using public API of the processor (`Register...` functions) to do the setup, that way you:
- Have better test coverage
- Are testing the object in a way it will be used. Arbitrary changes to internal state can sometimes lead to subtle bugs being missed. It doesn't seem to be the case here, but is a good principle to follow in general.

Reply:
Makes a lot of sense.